src/gleam/regex.gleam

```diff
@@ -3,8 +3,26 @@
 //// all of the PCRE library is interfaced and some parts of the library go beyond
 //// what PCRE offers. Currently PCRE version 8.40 (release date 2017-01-11) is used.
 
+import gleam/option.{Option}
+
 pub external type Regex
 
+/// The details about a particular match:
+///
+/// - match — the full string of the match.
+/// - index — the index of the match in the original string.
+/// - number — each match is numbered, starting from 1
+/// - submatches — a Regex can have subpatterns, sup-parts that are in parentheses.
+///
+pub type Match {
+  Match(
+    match: String,
+    index: Int,
+    number: Int,
```

Comment:
What is the number for?

Reply:
This is useful when paired with replace if replacement is dependent on how many times a pattern has appeared before.

I'll remove it from this PR and add it once we have a PR with replace functionality going.

Reply:
Could we use `index_map` for that instead? Seems odd to me that the index would be contained in the data structure.